tests/phpunit/tests/query/cacheResults.php

```diff
@@ -827,6 +845,72 @@ public function test_query_cache_delete_meta() {
 		$this->assertNotSame( $query1->found_posts, $query2->found_posts );
 	}
 
+
+	/**
+	 * @ticket 56802
+	 */
+	public function test_query_cache_like_meta() {
+		static $placeholder;
+		$old_placeholder = $placeholder;
+		$p1              = self::$posts[0];
+
+		$args   = array(
+			'cache_results' => true,
+			'fields'        => 'ids',
+			'meta_query'    => array(
+				array(
+					'key'     => 'color',
+					'value'   => '00',
+					'compare' => 'LIKE',
+				),
+			),
+		);
+		$query1 = new WP_Query();
+		$posts1 = $query1->query( $args );
+
+		// Force placeholder to be different.
+		$placeholder = rand();
+
+		$query2 = new WP_Query();
+		$posts2 = $query2->query( $args );
+
+		$placeholder = $old_placeholder;
+
+		$this->assertSame( $posts1, $posts2 );
+		$this->assertContains( $p1, $posts2 );
+		$this->assertSame( $query1->found_posts, $query2->found_posts );
+	}
+
+	/**
+	 * @ticket 56802
+	 */
+	public function test_query_cache_search() {
+		static $placeholder;
+		$old_placeholder = $placeholder;
```

Comment:
I'm not sure this is doing what you expect as the placeholder is a static within wpdb. Are these tests failing on trunk?

https://github.com/WordPress/wordpress-develop/blob/fed98bd9ef9a232d102c41e74944d3c21cd6183e/src/wp-includes/class-wpdb.php#L2419-L2440

Reply:
I hate static variables like this. Almost impossible to test. The only way I feel like this would be testable would be converting the static variable to a protected class property and then using reflections to change the property value. 

Unless you can think of clever to change the placeholder changing like on a new session. Thoughts?

Reply:
> Unless you can think of clever to change the placeholder changing like on a new session. Thoughts?

I'm quite stumped too. I'm wondering if the tests can reinitilize wpdb in the tests but I don't think that will work as the previous instance will have filters, etc that have been passed by reference. 

Modifying the signature for tests seems problematic to me as it will require updates from drop-in authors and may end up been abused in plugins.

Reply:
I have updated the tested and found a way to reset the placeholder. It is not the best as we add changing wpdb just for the tests. But there is no other way around it.